core/snapshot_creator.rs

```diff
@@ -0,0 +1,59 @@
+// Note: This is a nearly identical rewrite of core/libdeno/snapshot_creator.cc
+// but in Rust.
+//
+// This snapshot program is considered "basic" because the code being
+// snapshotted cannot call ops.
+
+#![allow(dead_code)]
+
+// TODO(ry) DRY up this repeated code from lib.rs
+
+#[macro_use]
+extern crate log;
+extern crate futures;
+extern crate libc;
+
+mod flags;
+mod isolate;
+mod js_errors;
+mod libdeno;
+mod module_specifier;
+mod modules;
+mod shared_queue;
+
+use crate::isolate::js_check;
+use crate::isolate::Isolate;
+use crate::isolate::StartupData;
+use std::env;
+use std::io::Write;
+
+fn main() {
+  let args: Vec<String> = env::args().collect();
+  // NOTE: `--help` arg will display V8 help and exit
+  let args = flags::v8_set_flags(args);
+
+  let (snapshot_out_bin, js_filename) = if args.len() == 3 {
+    (args[1].clone(), args[2].clone())
+  } else {
+    eprintln!("Usage: snapshot_creator <out_dir> <js_filename>");
```

Comment:
Looks like the 1st arg is a filename and not a dir?

Reply:
?

Reply:
fixed